.github/workflows/benchmark_on_push.yml

```
@@ -13,10 +13,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Set up Python 3.8
+      - name: Set up Python 3.11
```

```suggestion
      - name: Set up Python 3.12
```

Can we bump this to Python 3.12, it might be marginally faster and all dependencies support it?